### DISCUSSION TITLE
Using a custom property as options for an article, showing a dropdown list selection for end-user

### DISCUSSION THREAD

**Discussion by alex-pommier** [openstore-ecommerce/OpenStore#281] [Q&A]
I've created a custom property group called "Location" with GroupType "Property" and it holds a list of different locations that an article can be for. I would like to show to the end-user a dropdown list on the product detail page where they must choose one of these locations before adding-to-cart. Is this possible to do? If so can you offer any guidance.

Merci!

**Comment by DNNMonster** [comment]
If you assign it to a product the ProductDisplayDetail.cshtml file should render the property without any other work.

You should see the properties section under each product to let you add a property to a product.  I am guessing that you may not have a property yet associated with a product...  Check that first.

![open-store-properties](https://user-images.githubusercontent.com/12777553/191127443-6a819e76-2681-42dd-bdbb-ef2c7fd1cbf0.png)

**Reply by DNNMonster** [reply]
Err....  Actually I think what you may want is an option versus a property...  
![open-store-options](https://user-images.githubusercontent.com/12777553/191128349-84e1018c-9128-42b1-9754-fd6afdb7c537.png)

**Reply by alex-pommier** [reply]
I did apply the property to a product and selected some options, but on the front-end those options do not appear for a user to select from like IE: an option dropdown with some selection values to choose from.
![Backend](https://user-images.githubusercontent.com/13565738/191128440-03ec567c-6bd5-4dd6-93cd-01c02f78d8c1.png)
![frontend](https://user-images.githubusercontent.com/13565738/191128451-c3ffc6bf-d5de-4d97-bd1f-a7a188fd2688.png)

**Reply by alex-pommier** [reply]
> Err.... Actually I think what you may want is an option versus a property... ![open-store-options](https://user-images.githubusercontent.com/12777553/191128349-84e1018c-9128-42b1-9754-fd6afdb7c537.png)

Correct, the options would be ideal except I have a list of 100 option values that every product should have and adding those in manually for every product is a task i'd like to avoid if possible the properties are perfect because I can just choose which of the 100 options apply to that product instead of having to manually add the 20-40 options values that 1 product can have.

**Reply by alex-pommier** [reply]
I'm trying to do it via the XML file to load in the data as options vs properties but in order to make this viable for my client I would need to figure out a way of just letting them choose which property options apply to a article vs having to add them in as article options it's too tedious and they won't be able to manage XML.

**Reply by DNNMonster** [reply]
Are you importing products using the import tool?

**Reply by alex-pommier** [reply]
No i've added them through the interface and the exported to see if there's a quicker way of loading in all the option values directly in the XML and then re-importing. I'm looking through the database as well through SSMS but it looks like the data is stored in XML in the DB.

**Reply by DNNMonster** [reply]
Yep it is indeed XML for much of the data.  It taught json everything it knows.  :)  Then json fat shamed it.  Claimed it was too chatty.

I do bring in products with options using the Import Tool so it works but to have this process be dynamic I use a scheduled task that generates the Store Import XML and it relies on user preferences in the db created by a separate web ui.  It lets you use the DNN admin tools to manually run the process and/or automate it.  It's just one approach. 

Something similar should work for you and I don't think 100 options will be an issue...  but I have not tested it.

Xml can be a little painful but once you get the import for your system working you will not look at it too much more other than making an occasional adjustment.

I'm sure we'll get some input here if I'm overlooking something...
